### PR TITLE
Add minimum allowed amounts used in depositing and withdrawing from town and nation banks.

### PR DESCRIPTION
**Pull request by illumeleet**
#### Description: 

Add ability to set minimum deposit amount to towns & nations, also fixes some lang files malformations.
_accidentally got these two first commits, don't beat me pls_
____
#### New Nodes/Commands/ConfigOptions: 

town_min_deposit: '0'
town_min_withdraw: '0'
nation_min_deposit: '0'
nation_min_withdraw: '0'

____
#### Relevant Towny Issue ticket:

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by LlmDl** [comment]
The lang file version doesn't have to be bumped to .116 yet. Its also not recommended to add to all the lang files, if I do an update it means you have to rebase your langfiles to resolve the conflicts. I don't mind adding all the lang files after a PR is accepted.

**Comment by illumeleet** [comment]
well, sorry. how should i commit language file changes then? Also, do you approve changes made in TownCommand & NationCommand?

**Comment by Warriorrrr** [comment]
You can add your new lang strings just to the english.yml, makes rebasing less trivial. I'm pretty sure some economy plugins like TNE allow you to /pay directly into non-player accounts, which would bypass this setting.

File: src/com/palmergames/bukkit/towny/command/NationCommand.java [hunk 1]
```diff
@@ -2681,8 +2681,11 @@ private static void nationTransaction(Player player, String[] args, boolean with
 			if (args.length == 2) {
 				if (withdraw)
 					MoneyUtil.nationWithdraw(player, resident, resident.getTown().getNation(), amount);
-				else 
+				else {
+					double nationMinDeposit = TownySettings.getNationMinDeposit();
+					if (amount < nationMinDeposit) throw new TownyException(Translation.of("msg_err_deposit_must_be_higher_than", nationMinDeposit));
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/NationCommand.java:2686]: replace nationMinDeposit with `TownyEconomyHandler.getFormattedBalance(nationMinDeposit)`

File: src/com/palmergames/bukkit/towny/command/TownCommand.java [hunk 2]
```diff
@@ -3973,8 +3973,11 @@ private static void townTransaction(Player player, String[] args, boolean withdr
 
 				if (withdraw)
 					MoneyUtil.townWithdraw(player, resident, resident.getTown(), amount);
-				else
+				else {
+					double townMinDeposit = TownySettings.getTownMinDeposit();
+					if (amount < townMinDeposit) throw new TownyException(Translation.of("msg_err_deposit_must_be_higher_than", townMinDeposit));
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/TownCommand.java:3978]: replace townMinDeposit with `TownyEconomyHandler.getFormattedBalance(townMinDeposit)`

**Review by LlmDl — CHANGES_REQUESTED** [review]
Very close to perfect, change the two lines I specified to use the formatter, and set the lang files back to 0.115 and lose the newline and #Added in 0.116.

**Comment by illumeleet** [comment]
Done

**Review by LlmDl — APPROVED** [review]

**Comment by illumeleet** [comment]
> You can add your new lang strings just to the english.yml
Thank you, will do next time

> I'm pretty sure some economy plugins like TNE allow you to /pay directly into non-player accounts, which would bypass this setting.
Not sure about that, but PR references default Towny town&nation deposit commands. If there will be any errors in future, i am ready to contribute.

**Review by LlmDl — CHANGES_REQUESTED** [review]
It has dawned on me that the tests you added which run in TownCommand and NationCommand should actually be ran in the MoneyUtil's commonTests() method.

```java 
		if (!withdraw) {
			double minAmount = nation ? TownySettings.getNationMinDeposit() : TownySettings.getTownMinDeposit();
			if (amount < minAmount)
				throw new TownyException(Translation.of("msg_err_deposit_must_be_higher_than", TownyEconomyHandler.getFormattedBalance(minAmount)));
		}
```

**Comment by illumeleet** [comment]
Done

**Comment by illumeleet** [comment]
> commonTests()

Totally missed this one, if this implementation won't get approved, i'll move tests to commonTests.

**Review by LlmDl — APPROVED** [review]

**Pull request merged**